pkg/controller/ingressstate/ingress_state_controller.go

```diff
@@ -0,0 +1,328 @@
+package ingressstate
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	ingressStateControllerWorkQueueKey = "key"
+
+	maxToleratedPodPendingDuration = 5 * time.Minute
+)
+
+// degradedConditionNames contains all conditions that this controller manages
+var degradedConditionNames = sets.NewString(
```

Comment:
I guess `degradedConditionTypes` would be more accurate to stick with the conditions lingo

Reply:
Done